pandas/core/indexes/interval.py

```diff
@@ -699,8 +700,16 @@ def copy(self, deep=False, name=None):
 
     @Appender(_index_shared_docs['astype'])
     def astype(self, dtype, copy=True):
-        if is_interval_dtype(dtype):
-            return self.copy() if copy else self
+        dtype = pandas_dtype(dtype)
+        if is_interval_dtype(dtype) and dtype != self.dtype:
```

Comment:
you could do:`` not is_dtype(equal(dtype, self.dtype)``

Reply:
I think this needs to be a bit more specific than `not is_dtype_equal`; what this is really looking for is an `IntervalDtype` with a different subtype than the existing dtype.  Just having `not is_dtype_equal` would allow non-interval dtypes to pass through as well.

I could make the "different subtype" part more explicit with `and not is_dtype_equal(dtype.subtype, self.dtype.subtype)`.  Though this would be a partial dupe of the `IntervalDtype.__eq__` logic.  Could also just add a comment noting the "different subtype" part, if that would be preferable.

Reply:
I c, this is ok then. these are guaranteed to be II types and thus they are comparable